x-pack/filebeat/module/gsuite/_meta/docs.asciidoc

```diff
@@ -0,0 +1,102 @@
+- https://developers.google.com/admin-sdk/reports/v1/appendix/activity/user-accounts[User Accounts Activity Events]
+
+=== Configure the module
+
+To allow the filebeat module to ingest data from the Google Reports API, you would need an admin account, and set up a `ServiceAccount` to access `Admin SDK API`. Also notice, https://developers.google.com/admin-sdk/reports/v1/guides/delegation[Domain-Wide Delegation] is required for your application in order to work properly.
+
+This module will make use of the following `oauth2 scope`:
+
+- `https://www.googleapis.com/auth/admin.reports.audit.readonly`
+
+Once you have downloaded your service account credentials as a JSON file,
+you can set up your module:
+
+[float]
+===== Configuration options
+
+[source,yaml]
+----
+- module: gsuite
+  saml:
+    enabled: true
+    var.jwt_file: "./credentials_file.json"
+    var.delegated_account: "user@example.com"
+  user_accounts:
+    enabled: true
+    var.jwt_file: "./credentials_file.json"
+    var.delegated_account: "user@example.com"
+----
+
+Every fileset has the following configuration options:
+
+*`var.jwt_file`*::
+
+Specifies the path to the JWT credentials file.
+
+*`var.delegated_account`*::
+
+Email of the admin user used to access the API.
+
+*`var.http_client_timeout`*::
+
+Duration of the time limit on HTTP requests made by the module. Defaults to
+`60s`.
+
+*`var.interval`*::
+
+Duration between requests to the API. Defaults to `60s`.
+
+*`var.user_key`*::
+
+Specifies the user key to fetch reports from. Defaults to `all`.
+
+[float]
+==== GSuite Reports ECS fields
+
+This is a list of GSuite Reports fields that are mapped to ECS.
+
+[options="header"]
+|=======================================================================================
+| GSuite Reports             | ECS Fields                                              |
+| items[].id.time            | @timestamp                                              |
+| items[].id.uniqueQualifier | event.id                                                |
+| items[].id.applicationName | event.provider                                          |
+| items[].events[].name      | event.action                                            |
+| items[].customerId         | organization.id                                         |
+| items[].ipAddress          | client.ip, related.ip, client.as.*, client.geo.*        |
+| items[].actor.email        | client.user.email, client.user.name, client.user.domain |
+| items[].actor.profileId    | client.user.id                                          |
+|=======================================================================================
+
+These are the common ones to all filesets.
+
+[float]
+=== Example dashboard
+
+This module comes with a sample dashboard. For example:
+
+TODO: include an image of a sample dashboard. If you do not include a dashboard,
```

Comment:
Is this TODO still valid or does this and the surrounding few lines need to be removed?

Reply:
Plan to add a dashboard later, but will remove the section in the meanwhile :+1: